kairon/shared/data/processor.py

```
@@ -9241,9 +9246,68 @@
+
+        raw_params = dict(request.query_params)
+        valid_fields = MongoProcessor.get_field_ids_for_log_type(log_type)
+
+        sanitized = {}
+
+        for k, v in raw_params.items():
+            if k in {"from_date", "to_date"}:
+                try:
+                    sanitized[k] = date.fromisoformat(v)
+                except ValueError:
+                    raise ValueError(f"Invalid date format for '{k}': '{v}'. Use YYYY-MM-DD.")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)